docs/guides/async.mdx

```diff
@@ -0,0 +1,188 @@
+---
+title: Async
+description: This doc describes about the behavior with async.
+nav: 1.03
+---
+
+Using async atoms, you gain access to real-world data while still managing them directly from your atoms and with incredible ease.
+
+We can seperate them in two main categories:
+
+- Async read atoms: async request is started instantly as soon as you try to get its value, you could relate to them as "smart getters"
+- Async write atoms: async request is started at a specific moment, you could relate to them as "actions"
+
+## Async read atom
+
+The `read` function of an atom can return a promise.
+
+```js
+const countAtom = atom(1)
+const asyncAtom = atom(async (get) => get(countAtom) * 2)
+```
+
+Jotai is inherently leveraging `Suspense` to handle asynchronous flows.
+
+```jsx
+const ComponentUsingAsyncAtoms = () => {
+  const [num] = useAtom(asyncAtom)
+  // here `num` is always `number` even though asyncAtom returns a Promise
+}
+const App = () => {
+  return (
+    <React.Suspense fallback={/* What to show while suspended */}>
+      <ComponentUsingAsyncAtoms />
+    </React.Suspense>
+  )
+}
+```
+
+Alternatively, you could avoid the inherent suspending that jotai does for you, by wrapping your atoms with the [`loadable` API](../utils/loadable.mdx).
+
+**Note**: An atom becomes async not only if the atom read function is async,
+but also if one or more of its dependencies are async.
+
+```js
+const anotherAtom = atom((get) => get(asyncAtom) / 2)
+// even though this atom doesn't return a promise,
+// it's a read async atom because `asyncAtom` is async.
+```
+
+**Note**: You cannot get the value of an async atom in a write-atom (wether its async or not) before its value has been set
```

Comment:
This only applies to derived atoms. primitive atoms such as `atom(1)` should be fine.

Reply:
@dai-shi I'm not sure I understand your statement.

To me, the error appears as soon as its an `async atom`, not `derived atom`
I was trying to talk about `async` atoms (`atom(1)` is not async).

Maybe there's something I'm missing?
Do you think I should remove this part?

Reply:
Oh, you are right. I was confused with "async or not" part.
```suggestion
**Note**: You cannot get the value of an async atom in a write-atom (whether the write function is async or not) before its value has been resolved
```